Content.Server/_Gabystation/ATM/ATMSystem.cs

```diff
@@ -0,0 +1,190 @@
+// SPDX-FileCopyrightText: 2025 AgentePanela <agentepanela@gmail.com>
+// SPDX-FileCopyrightText: 2025 GabyChangelog <agentepanela2@gmail.com>
+//
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+using Content.Server._Gabystation.Economy;
+using Content.Shared._Gabystation.ATM;
+using Content.Shared._Gabystation.NanoBank;
+using Content.Shared.Cargo.Components;
+using Content.Shared.Stacks;
+using Content.Shared.UserInterface;
+using Robust.Server.Audio;
+using Robust.Server.GameObjects;
+using Robust.Shared.Containers;
+
+namespace Content.Server._Gabystation.ATM;
+
+public sealed partial class BankATMSystem : SharedBankATMSystem
+{
+    [Dependency] private readonly UserInterfaceSystem _uiSystem = default!;
+    [Dependency] private readonly EconomyManagerSystem _economy = default!;
+    [Dependency] private readonly AudioSystem _audio = default!;
+    [Dependency] private readonly SharedStackSystem _stack = default!;
+
+    public override void Initialize()
+    {
+        SubscribeLocalEvent<BankATMComponent, BeforeActivatableUIOpenEvent>(UpdateInterfaceEvent);
+        SubscribeLocalEvent<BankATMComponent, EntInsertedIntoContainerMessage>(UpdateInterfaceEvent);
+        SubscribeLocalEvent<BankATMComponent, EntRemovedFromContainerMessage>(UpdateInterfaceEvent);
+
+        Subs.BuiEvents<BankATMComponent>(BankATMUIKey.Key, subs =>
+        {
+            subs.Event<BankATMMessage>(OnMesssage);
+        });
+
+        base.Initialize();
+    }
+
+    public override void Update(float frameTime)
+    {
+        base.Update(frameTime);
+
+        var ents = AllEntityQuery<BankATMComponent>();
+        while (ents.MoveNext(out var uid, out var comp))
+        {
+            if (comp.Printing >= 0f)
+            {
+                comp.Printing -= frameTime;
+                continue;
+            }
+            if (comp.MoneyToPrint <= 0)
+                continue;
+
+            if (!TryComp(uid, out TransformComponent? xform))
+                continue;
```

Comment:
```suggestion
            var xform = Transform(uid);
```

Estamos no servidor, essa entidade com certeza é válida e logo tem o componente `TransformComponent`.